File: 50 Coding Challenge Part II/Python/Challenge32.py
# Challenge 32: Create a function that will capitalize the first letter of each
# word in a text

# Using the capitalize()
def capitalizeWord(text):
    arr = text.split()
    result = list()
    for i in arr:
        result.append(i.capitalize())
    return " ".join(result)


txt = "hello world"
print(capitalizeWord(txt))

File: 50 Coding Challenge Part II/TypeScript/Challenge32.ts
// Challenge 32:Create a function that will capitalize the first letter of each
// word in a text

// toTitleCase function transform a string of text to titleCase
function toTitleCase (text: string): string {
    let words: string[] = text.split(" ");
    for (let i = 0; i < words.length; i++){
        let newWord = `${words[i].slice(0, 1).toUpperCase()}${words[i].slice(1, words[i].length).toLowerCase()}`
        words[i] = newWord;
    }
    return words.join(" ");
}

// Driver Method
let str = "my dear friends";
console.log(toTitleCase(str));
